laserscan_toolbox_node:
  ros__parameters:
    #common
    rate: 20.0 #hz
    topic_out: "scan" 
    new_frame: "base2_link"

    #debugging
    debug: true
    debug_file_path: "/home/jrluser/Desktop/call_m_workspace/lasertoolbox_debug.txt"
    show_ranges: false

    #output settings
    angle_min: 0.0
    angle_max: 6.283185307
    angle_increment: 0.017453293 #0.017453293 for 360 values resolution
    range_min: 0.0 #m
    range_max: 12.0 #m

    #advanced output settings
    auto_set: true #If true, the seetings below will be set automatically using the first source
    time_increment: 0.0 #s
    scan_time: 0.0 #s

    #sources
    sources: lidar1 lidar2

    lidar1:
      topic: "lidar1/scan"
      start_angle: 0.785398163 #45 deg
      end_angle: 5.497787144 #-45 = 315 deg
      scan_angle_offset: 3.141592654 #180 deg with rpLidars
      range_min: 0.0 #m
      range_max: 100.0 #m
      persistence: false
      timeout: 0.0 #0.0 seconds = all value's date are autorized

    lidar2:
      topic: "lidar2/scan"
      start_angle: 0.785398163 #45 deg
      end_angle: 5.497787144 #-45 = 315 deg
      scan_angle_offset: 3.141592654 #180 deg with rpLidars
      range_min: 0.0 #m
      range_max: 100.0 #m
      persistence: false
      timeout: 0.0 #0.0 seconds = all value's date are autorized